### PR TITLE
Added tests (and bug fixes)

### LINKED ISSUE #57: Add integration tests to Alexa Skill

### LINKED ISSUE #79: Implement Travis for Alexa skill

### PR DESCRIPTION
**Pull request by BScong**
Fixes #57 
Fixes #79 

Adding tests to Alexa Skill using Bespoken virtual-alexa.
Use of two different accounts for testing: scoutskilltest@mailinator.com and scoutskilltest2@mailinator.com. At the moment the second one is only useful to have coverage over the case where the number of articles is a multiple of TITLE_CHUNK_LEN.

Current coverage is **88%.**

Before merging:

- Register both tests users in scout-stage.

Notes:

- When testing if the skill is playing a file, we just test that the Alexa is playing something (not especially the right file). That can be quite hard to implement since the file names are dynamic.

- We have a portion of code in audio_controller.js that might never execute. We have to investigate that. A branch of state_handlers.js also seem to never execute.

- We currently don't catch the Scout Unavailable rejection.

- SearchAndSummarizeArticle Intent is implemented in state_handlers.js but not declared in the model. So we are currently never executing this code.

- Yes and No intents don't make sense in our current implementation. They are not tested.

- Pause and Resume intents are implemented in every state. But they should only be used in the Play Mode state.

- We also lack in test coverage because cases when the API (scout-ua) is down are not currently tested.

- It seems like some test cases are not reproducible in real (like PlayMode - Play different article).

- Strange behavior of virtual-alexa when doing actions while playing audio (such as asking to play a different file while playing).

- As we are doing a lot of calls to the Pocket API in a very short interval of time, we could be rate limited when running the tests twice in a row.

### CODE REVIEW

File: lambda/custom/test/tests.js [hunk 1]
```diff
@@ -0,0 +1,866 @@
+var AWS = require('aws-sdk');
+AWS.config.update({ region: 'us-east-1' });
+const assert = require('chai').assert;
+const vax = require('virtual-alexa');
+const constants = require('../constants');
+// You need to setup the same ENV var as the Lambda function
```
Comment (tamarahills) [inline lambda/custom/test/tests.js]: I think we should document this note in readme perhaps?

File: lambda/custom/test/tests.js [hunk 2]
```diff
@@ -0,0 +1,866 @@
...
+const alexa = vax.VirtualAlexa.Builder()
+  .handler('index.handler') // Lambda function file and name
+  .interactionModelFile('../../models/en-US.json')
+  .create();
+alexa.context().setAccessToken('scoutskilltest@mailinator.com');
+
+const checkStreamingString = '.mp3';
+
+describe('Unit Tests', function() {
```
Comment (tamarahills) [inline lambda/custom/test/tests.js]: We should call these integration tests instead of unit tests.

File: lambda/custom/test/tests.js [hunk 3]
```diff
@@ -0,0 +1,866 @@
...
+    let result = await alexa.launch();
+    result = await alexa.utter('Stop');
+    assert.include(result.prompt(), constants.strings.ALEXA_STOP_RESP);
+  });
+  it('Pause', async () => {
+    let result = await alexa.launch();
+    result = await alexa.utter('Pause');
+    assert.isNotTrue(alexa.audioPlayer().isPlaying());
+  });
+  /*it('Resume', async () => {
```
Comment (tamarahills) [inline lambda/custom/test/tests.js]: Should we remove this? Or create a follow-on patch to get this to work?

File: lambda/custom/test/tests.js [hunk 4]
```diff
@@ -0,0 +1,866 @@
...
+        checkStreamingString
+      );
+      let firstArticle = result.response.directives[0].audioItem.stream.url;
+      assert.isTrue(alexa.audioPlayer().isPlaying());
+
+      result = await alexa.utter('Play Donald Trump');
+      //assert.isNotTrue(alexa.audioPlayer().isPlaying());
+      assert.include(result.prompt(), constants.strings.TITLE_CHOOSE_SUMM_FULL);
+      result = await alexa.utter('full article');
+      /*assert.include(
```
Comment (tamarahills) [inline lambda/custom/test/tests.js]: Can we remove or create a follow up?

File: lambda/custom/test/tests.js [hunk 5]
```diff
@@ -0,0 +1,866 @@
...
+      result = await alexa.utter('full article');
+      assert.include(
+        result.response.directives[0].audioItem.stream.url,
+        checkStreamingString
+      );
+      let firstArticle = result.response.directives[0].audioItem.stream.url;
+      assert.isTrue(alexa.audioPlayer().isPlaying());
+
+      result = await alexa.utter('Play Donald Trump');
+      //assert.isNotTrue(alexa.audioPlayer().isPlaying());
```
Comment (tamarahills) [inline lambda/custom/test/tests.js]: can we remove or create a follow up?

File: lambda/custom/test/tests.js (hunk 1 above)
Comment (BScong) [reply lambda/custom/test/tests.js]: Added. Also added an error message

**Pull request merged**